### PR TITLE
chore: Upgrade Node.js to v20 (LTS)

### PR DESCRIPTION
**Pull request by aineo**
- chore: upgrade node js to version 20 in CI/CD
- chore: upgrade github actions checkout version to 4
- chore: upgrade github upload artifacts version to 4
- fix: high vulnerabilities
- chore: update `vite-plugin-pwa` to replace deprecated `rollup-plugin-terser`

### CODE REVIEW

**Comment by aineo** [comment]
[Upgrade Node.js to v20 (LTS)](https://trello.com/c/Tj1T5DaY/529-upgrade-nodejs-to-v20-lts)

File: .github/workflows/electron-linux.yml [hunk 1]
```diff
@@ -18,9 +18,9 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Setup Node version 🍀
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
```
Comment (bludnic) [inline .github/workflows/electron-linux.yml:21]: Check if there is a new version for other actions as well:

- actions/checkout@v3 -> v4
- actions/upload-artifact@v3 -> v4

https://github.com/actions/checkout
https://github.com/actions/upload-artifact

**Review by bludnic — CHANGES_REQUESTED** [review]

File: .github/workflows/electron-linux.yml (hunk 1 above)
Comment (aineo) [reply .github/workflows/electron-linux.yml:21]: I add changes in separate commits)

**Comment by aineo** [comment]
[Replace deprecated packages](https://trello.com/c/ojIllQ9b/550-replace-deprecated-packages)

**Comment by aineo** [comment]
[Fix Github Actions deprecations](https://trello.com/c/E2OeEsd2/541-fix-github-actions-deprecations)

**Review by bludnic — APPROVED** [review]

**Pull request merged**